### PR TITLE
Implemented F2 rename branch in RepoObjectsTree.

### LINKED ISSUE #8599: F2 key should open branch rename window from the branch list
## Feature description
F2 key should open branch rename window from the branch list on the left.
Context menu should be updated with text: "Rename (F2)"

## Environment
- Git Extensions 3.4.3.9999
- Build d4b0f48bbf3e39a71f5d7ff5231be5678d4aa0f1
- Git 2.28.0.windows.1
- Microsoft Windows NT 10.0.18363.0
- .NET Framework 4.8.4250.0
- DPI 96dpi (no scaling)

### PR DESCRIPTION
**Pull request by bcolaco**
Fixes #8599

## Proposed changes

- Added listener for F2 key in RepoObjetsTree that fired the branch rename dialog.

# To do

- Unit tests (maybe). I think a some refactoring of the existing code is required to cover the added code with unit tests.
- Change de hard coded F2 with the configured key in `HotkeySettingsManager`.
- Add the configured hot key text to the context menu item.

## Test methodology

- Manual

## Test environment(s)

- GIT 2.29.2.windows.3
- Windows 10.0.17763.0

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

**Comment by mstv** [comment]
I do not insist on using the `HotkeySettingsManager` since `F3` was hard-coded for searching, too, and `F2` is the standard for rename. Thoughts, @gitextensions/git-extensions-team?

File: GitUI/BranchTreePanel/RepoObjectsTree.cs [hunk 1]
```diff
@@ -526,7 +526,11 @@ private void TxtBranchCriterion_KeyDown(object sender, KeyEventArgs e)
 
         private void OnPreviewKeyDown(object sender, PreviewKeyDownEventArgs e)
         {
-            if (e.KeyCode == Keys.F3)
+            if (e.KeyCode == Keys.F2)
```
Comment (RussKie) [inline GitUI/BranchTreePanel/RepoObjectsTree.cs:529]: Can we use a `switch` statement here?

Comment (bcolaco) [reply GitUI/BranchTreePanel/RepoObjectsTree.cs:529]: Yes, I think it makes sense, especially if the list of hotkeys handled tends to grow,

**Review by drewnoakes — APPROVED** [review]

File: GitUI/BranchTreePanel/RepoObjectsTree.cs [hunk 2]
```diff
@@ -526,9 +526,14 @@ private void TxtBranchCriterion_KeyDown(object sender, KeyEventArgs e)
...
         {
-            if (e.KeyCode == Keys.F3)
+            switch (e.KeyCode)
             {
-                OnBtnSearchClicked(null, null);
+                case Keys.F2:
+                    Node.OnNode<Node>(treeMain.SelectedNode, node => node.OnRename());
+                    break;
+                case Keys.F3:
+                    OnBtnSearchClicked(null, null);
```
Comment (RussKie) [inline GitUI/BranchTreePanel/RepoObjectsTree.cs:535]: We should call `DoSearch();` instead.
We shouldn't be calling `OnBtnSearchClicked(null, null);` like we are now.

**Review by RussKie — CHANGES_REQUESTED** [review]
I don't believe the proposed change is correct - it is unnecessary excessive.

I believe the following is sufficient:
```diff
diff --git a/GitUI/BranchTreePanel/RepoObjectsTree.cs b/GitUI/BranchTreePanel/RepoObjectsTree.cs
index caec94b90..487d85f6c 100644
--- a/GitUI/BranchTreePanel/RepoObjectsTree.cs
+++ b/GitUI/BranchTreePanel/RepoObjectsTree.cs
@@ -529,9 +529,17 @@ private void TxtBranchCriterion_KeyDown(object sender, KeyEventArgs e)
 
         private void OnPreviewKeyDown(object sender, PreviewKeyDownEventArgs e)
         {
-            if (e.KeyCode == Keys.F3)
+            switch (e.KeyCode)
             {
+                case Keys.F2:
+                    {
+                        Node.OnNode<LocalBranchNode>(treeMain.SelectedNode, node => node.Rename());
+                        break;
+                    }
+
+                case Keys.F3:
                     OnBtnSearchClicked(null, null);
+                    break;
             }
         }
 ```

**Comment by RussKie** [comment]
Please provide your sign off to the contributors.txt.

**Comment by GintasS** [comment]
This is an old PR and should be closed or set to a draft. @RussKie

**Comment by RussKie** [comment]
@bcolaco please rebase and we can merge this.

**Comment by bcolaco** [comment]
I think I'll be able to do it next week.

**Comment by bcolaco** [comment]
@RussKie rebase is complete.

**Review by gerhardol — COMMENTED** [review]
A mnemonics should be added to the menu too (CreateRename())

**Review by RussKie — APPROVED** [review]

**Comment by RussKie** [comment]
Please add the mnemonic too.

**Comment by bcolaco** [comment]
I'm not very happy with the mnemonic implementation: I think the cast doesn't look very good and I don't like the hard coded `Keys.F2`.
Any suggestions on how to improve this?

**Comment by gerhardol** [comment]
> I'm not very happy with the mnemonic implementation: I think the cast doesn't look very good and I don't like the hard coded `Keys.F2`.
> Any suggestions on how to improve this?

```
                ToolStripMenuItem item = _menuItemFactory.CreateMenuItem<ToolStripMenuItem, TNode>(
                                    node => ((ICanRename)node).Rename(), Strings.Rename, GetTooltip(MenuItemKey.Rename), Properties.Images.Renamed.AdaptLightness());
                item.ShortcutKeys = Keys.F2;
                yield return item.WithKey(MenuItemKey.Rename);
```

The hardcoding itself is a little ugly too, not much to do about that.

**Review by gerhardol — APPROVED** [review]

**Comment by RussKie** [comment]
Thank you

**Pull request merged**